src/System.Management.Automation/engine/MshMemberInfo.cs

```diff
@@ -4931,15 +4934,46 @@ public override IEnumerator<T> GetEnumerator()
             return new Enumerator<T>(this);
         }
 
-        internal override T GetFirstOrDefault(MemberNamePredicate predicate)
+        internal override T FirstOrDefault(MemberNamePredicate predicate)
         {
+            object delegateOwner;
+            if (_mshOwner != null)
+            {
+                delegateOwner = _mshOwner;
+                foreach (PSMemberInfo member in _mshOwner.InstanceMembers)
+                {
+                    if (member is T memberAsT && predicate(memberAsT.Name))
+                    {
+                        return memberAsT;
+                    }
+                }
+            }
+            else
+            {
+                delegateOwner = _memberSetOwner.instance;
+                foreach (PSMemberInfo member in _memberSetOwner.InternalMembers)
+                {
+                    if (member is T memberAsT && predicate(memberAsT.Name))
+                    {
+                        memberAsT.ReplicateInstance(delegateOwner);
+                        return memberAsT;
+                    }
+                }
+            }
+
+            if (delegateOwner == null)
+            {
+                return null;
+            }
```

Comment:
Handle `PSObject` instance members and `PSMemberSet` internal members here.
Your original implementation in `PSObject.GetFirstPropertyOrDefault` handles instance members, but that doesn't cover the `PSMemberSet` case in this method. So it's probably better to handle both here.